haroldtreen/readmeio-sync
# Code review cleanup + Upload Using Diffs

# Code Review Cleanup + Upload Using Diffs

Status: **Ready for Review**

Reviewers: @fractaltheory @donnielrt 
JIRA: [RTM-224](https://mobify.atlassian.net/browse/RTM-224)
Linked PRs: [Remote Clean Funtionality](https://github.com/mobify/readmeio-sync/pull/4), (Code Review)[https://github.com/mobify/readmeio-sync/pull/2]
## Changes
- Created download object for fetching the remote registry
- Added Async library to do network requests in parallel
- Cleaned lib/config.js
- Cleaned lib/contentExtractor.js
- Cleaned lib/registry.js
- Cleaned lib/cli.js
- Cleaned lib/requestor.js
## Test Drive
- Clone this repository
- npm link inside it
- Set README_EMAIL, README_PASSWORD
- Create a config with `readmeio-sync config`
- `readmeio-sync init` (make sure production project is used)
- `readmeio-sync clean-remote` (make sure all content not specified in registry is deleted from staging)
- `readmeio-sync upload` (make sure all content in registry is uploaded to staging)
## Research Resource
- [README](https://github.com/mobify/readmeio-sync/blob/code-review-cleanup/README.md)
